### PR TITLE
Support v2.1 and v2.2

### PR DESCRIPTION
**Pull request by 0x0Louis**
Support v2.1 and v2.2

### CODE REVIEW

File: src/LBQuoter.sol [hunk 1]
```diff
@@ -147,18 +171,55 @@ contract LBQuoter {
...
+                                        swapForY
+                                    );
+
+                                    quote.fees[i] = ((uint256(fees) * 1e18) / quote.amounts[i]).safe128(); // fee percentage in amountIn
+                                }
+                            } catch {}
+                        }
+                    }
+                }
+            }
```
Comment (Mathieu-Be) [inline src/LBQuoter.sol:208]: This code is the same for 2.1 and 2.2 right ?

File: src/interfaces/ILBRouter.sol [hunk 2]
```diff
@@ -52,7 +52,8 @@ interface ILBRouter {
     enum Version {
         V1,
         V2,
-        V2_1
+        V2_1,
+        V2_2
     }
```
Comment (Mathieu-Be) [inline src/interfaces/ILBRouter.sol:57]: natspec needs to be updated

**Review by pecio222 — APPROVED** [review]
maybe worth adding that LBQuoter is gas intensive and not meant to query on-chain

File: src/LBQuoter.sol (hunk 1 above)
Comment (0x0Louis) [reply src/LBQuoter.sol:208]: Yes, the ABIs are the same for this reason

Comment (0x0Louis) [reply src/LBQuoter.sol:208]: It could be abstracted better in a separate functions, but I don't think it's worth changing the quoter

**Pull request merged**